### PR TITLE
[DWARFLinker][NFC] Rename libraries to match with directories name.

### PR DESCRIPTION
**Pull request by avl-llvm**
It was noted that new DWARFLinker libraries do not follow naming agreement - https://github.com/llvm/llvm-project/pull/75925#issuecomment-1883301659 This patch rename libraries to match with the agreement.

Rename LLVMDWARFLinkerBase library into the LLVMDWARFLinker. Rename LLVMDWARFLinker library into the LLVMDWARFLinkerClassic. Correct include path according to the new directory structure.

### CODE REVIEW

File: llvm/lib/DWARFLinker/Classic/CMakeLists.txt [hunk 1]
```diff
@@ -1,11 +1,11 @@
-add_llvm_component_library(LLVMDWARFLinker
+add_llvm_component_library(LLVMDWARFLinkerClassic
   DWARFLinkerCompileUnit.cpp
   DWARFLinkerDeclContext.cpp
   DWARFLinker.cpp
   DWARFStreamer.cpp
 
   ADDITIONAL_HEADER_DIRS
-  ${LLVM_MAIN_INCLUDE_DIR}/llvm/DWARFLinker
+  ${LLVM_MAIN_INCLUDE_DIR}/llvm/DWARFLinker/Classic
```
Comment (nico) [inline llvm/lib/DWARFLinker/Classic/CMakeLists.txt:8]: unrelated; maybe this line isn't needed at all?

File: llvm/lib/DWARFLinker/Parallel/CMakeLists.txt [hunk 2]
```diff
@@ -12,7 +12,7 @@ add_llvm_component_library(LLVMDWARFLinkerParallel
   SyntheticTypeNameBuilder.cpp
 
   ADDITIONAL_HEADER_DIRS
-  ${LLVM_MAIN_INCLUDE_DIR}/llvm/DWARFLinkerParallel
+  ${LLVM_MAIN_INCLUDE_DIR}/llvm/DWARFLinker/Parallel
```
Comment (nico) [inline llvm/lib/DWARFLinker/Parallel/CMakeLists.txt:15]: same comment – since things worked with the old, wrong, name, maybe we can toss it altogether? (but seems like something for a different pr)

File: llvm/unittests/DWARFLinkerParallel/CMakeLists.txt [hunk 3]
```diff
@@ -1,4 +1,5 @@
 set(LLVM_LINK_COMPONENTS
+  DWARFLinker
```
Comment (nico) [inline llvm/unittests/DWARFLinkerParallel/CMakeLists.txt:2]: also seems unrelated

**Review by nico — APPROVED** [review]
Thank you! Some optional nits below.

File: llvm/lib/DWARFLinker/Classic/CMakeLists.txt (hunk 1 above)
Comment (avl-llvm) [reply llvm/lib/DWARFLinker/Classic/CMakeLists.txt:8]: Ok, will change this and next comment with a separate patch.

File: llvm/unittests/DWARFLinkerParallel/CMakeLists.txt (hunk 3 above)
Comment (avl-llvm) [reply llvm/unittests/DWARFLinkerParallel/CMakeLists.txt:2]: This is necessary in general even if everything is working currently without DWARFLinker(as LLVMDWARFLinker contains base functionality).

**Comment by nico** [comment]
Looks good to me. I'll wait a day for Adrian or Jonas to shout, else I'll hit the merge button tomorrow afternoon, thanks :)

**Comment by avl-llvm** [comment]
Thank you for the review!
